tests/baselines/reference/lateBoundElementAccessAssignmentDeclarationsInJs.js

```diff
@@ -0,0 +1,29 @@
+//// [file.js]
+export function foo() {}
+foo.bar = 12;
+const _private = Symbol();
+foo[_private] = "ok";
+const strMem = "strMemName";
+foo[strMem] = "ok";
+const dashStrMem = "dashed-str-mem";
+foo[dashStrMem] = "ok";
+const numMem = 42;
+foo[numMem] = "ok";
+
+/** @type {string} */
+const x = foo[_private];
+/** @type {string} */
+const y = foo[strMem];
+/** @type {string} */
+const z = foo[numMem];
+/** @type {string} */
+const a = foo[dashStrMem];
+
+
+
+//// [file.d.ts]
+export function foo(): void;
+export namespace foo {
+    export const bar: number;
+    export const strMemName: string;
+}

```

Comment:
Is there a reason we couldn’t generate

```ts
export const foo: {
  (): void;
  bar: number;
  strMemName: string;
  "dashed-str-mem": string;
  [42]: string;
};
```

?

Reply:
Hmm, we could; I think I already have the bail path for that setup, and just need to adjust the conditions for it to occur. Thing is, that has the potential to break, too - if one of my members was an alias, eg `foo.Class = ClassRef`, the _only_ way to preserve that alias is by emitting it as an alias in a namespace, which can't then merge with a `const`. So it's like aliases and non-identifier names are mutually exclusive, syntactically T.T